src/libs/data-utils.js

```diff
@@ -340,3 +355,249 @@ export const EntityUploader = class EntityUploader extends Component {
     ])])
   }
 }
+
+validate.validators.duplicateEntityNameValidator = (newName, { takenName }) => {
+  if (takenName === newName) {
+    return 'An entity with this name already exists'
+  }
+}
```

Comment:
This adds a new global validator, which seems unneeded. Using the built-in `exclusion` with a custom message is probably a better option.